### PR TITLE
Added servo component to the registry

### PR DESCRIPTION
**Pull request by pedrominatel**
## Description

This PR adds the servo to the ESP Component Registry, including an example.

## Related

close https://github.com/espressif/esp-iot-solution/pull/419

## Testing

N/A

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [ ] Documentation is updated as needed.
- [ ] Tests are updated or added as necessary.
- [ ] Code is well-commented, especially in complex areas.
- [ ] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

**Comment by leeebo** [comment]
Hi @pedrominatel Thanks for your contribution!

**Review by YanKE01 — COMMENTED** [review]
Thank you for your contribution. We've made some suggestions in response to your submission to guarantee that the style of the component is consistent with the repository. In addition, we suggest adding the **servo** component to the CI, you can refer to this [link](https://github.com/espressif/esp-iot-solution/blob/master/.gitlab/ci/build.yml#L1167).

**Comment by pedrominatel** [comment]
Thank you very much @YanKE01 for your review and thoughts.
Please take a look at the latest changes.

File: .gitlab/ci/build.yml [hunk 1]
```diff
@@ -574,6 +574,18 @@ build_example_motor_foc_knob:
   variables:
     EXAMPLE_DIR: examples/motor/foc_knob_example
 
+build_example_motor_servo:
```
Comment (YanKE01) [inline .gitlab/ci/build.yml:577]: In addition to adding the servo component in `build.yml`, it is also necessary to add rules for servo in` rules.yml` to ensure that changes to servo and projects dependent on servo can trigger the CI. You can refer to any existing component in `rules.yml` as a reference to complete the rule definition for servo.

**Review by YanKE01 — COMMENTED** [review]
Hi, @pedrominatel . Thanks for your contribution. Some `CI rules` and `component specifications` still need your attention, and we've provided some reference suggestions.

File: .gitlab/ci/build.yml (hunk 1 above)
Comment (pedrominatel) [reply .gitlab/ci/build.yml:577]: I did but I'm not sure if I did correctly. Can you kindly review it?
Thank you, @YanKE01.

**Comment by YanKE01** [comment]
LGTM! Hi, @leeebo , can you help review this again? Thank you.

File: components/motor/servo/examples/servo_control/main/servo_control.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            vTaskDelay(20 / portTICK_PERIOD_MS);
+        }
+        // Return to the initial position
+        iot_servo_write_angle(LEDC_LOW_SPEED_MODE, 0, calibration_value_0);
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+    vTaskDelete(NULL);
+}
+
+void servo_init(void)
```
Comment (leeebo) [inline components/motor/servo/examples/servo_control/main/servo_control.c:36]:
```suggestion
static void servo_init(void)
```

add static as you only used in this file

File: components/motor/servo/examples/servo_control/main/servo_control.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#include "esp_log.h"
+#include "esp_system.h"
+#include "sdkconfig.h"
+#include "iot_servo.h"
+
+#define SERVO_GPIO          (2)  // Servo GPIO
+
+static const char *TAG = "Servo Control";
+
+uint16_t calibration_value_0 = 30; // Real 0 degree angle
```
Comment (leeebo) [inline components/motor/servo/examples/servo_control/main/servo_control.c:15]:
```suggestion
static uint16_t calibration_value_0 = 30; // Real 0 degree angle
```

**Comment by leeebo** [comment]
@pedrominatel LGTM! Thanks. Let's sync the PR first to run the CI

**Comment by leeebo** [comment]
sha=9eda883f174cda110b084a4cd1726fb336bc4e74

**Comment by leeebo** [comment]
hi @pedrominatel please install pre-commit to the current repo using `pre-commit install`, there are some files miss the `Check copyright notices`

**Review by leeebo — APPROVED** [review]

**Comment by pedrominatel** [comment]
Thank you @leeebo.

**Comment by leeebo** [comment]
Hi @pedrominatel , the github action `
Sync approved PRs to internal codebase`  failed with `RuntimeError: PR modifying forbidden files!!!`, the internal MR not be created.

Please manually create it.

1. add internal Gitlab repo to git remote
2. push to Gitlab then create MR
3. add `close https://github.com/espressif/esp-iot-solution/pull/419` in the commit message to link this PR

**Comment by pedrominatel** [comment]
> Hi @pedrominatel , the github action ` Sync approved PRs to internal codebase` failed with `RuntimeError: PR modifying forbidden files!!!`, the internal MR not be created.
> 
> Please manually create it.
> 
>     1. add internal Gitlab repo to git remote
> 
>     2. push to Gitlab then create MR
> 
>     3. add `close https://github.com/espressif/esp-iot-solution/pull/419` in the commit message to link this PR

Done.

**Pull request closed without merging**